lib/parsed_date_string.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class ParsedDateString
+  def initialize(date_string:, date_regex: /(?<day>\d+) de (?<month>[^[:space:]]*) de (?<year>\d+)/)
+    @date_string = date_string
+    @date_regex = date_regex
+  end
+
+  def to_s
+    return '' if match.nil?
+    '%d-%02d-%02d' % [match[:year], month(match[:month]), match[:day]]
+  end
+
+  private
+
+  attr_reader :date_string, :date_regex
+
+  def match
+    @match ||= date_string.match(date_regex)
+  end
+
+  def month(str)
+    ['', 'enero', 'febrero', 'marzo', 'abril', 'mayo', 'junio', 'julio', 'agosto', 'septiembre', 'octubre', 'noviembre', 'diciembre'].find_index(str.downcase) || raise("Unknown month #{str}".magenta)
```

Comment:
I very much doubt we actually want that `.magenta` there — we should be getting rid of `colorize` where possible, as it doesn't really play well with Morph.